spec/index.bs

```diff
@@ -1493,6 +1497,103 @@ interface GPUTextureUsage {
+
+    </dfn>
+  </div>
+
+  <div algorithm class=validusage>
+    <dfn abstract-op>createTexture Valid Usage</dfn>
+      Given a {{GPUDevice}} |this| and a {{GPUTextureDescriptor}} |descriptor|
+      the following validation rules apply:
+
+      1. |this| must be a [=valid=] {{GPUDevice}}.
+      1. |descriptor|.{{GPUTextureDescriptor/size}}.[=Extent3D/width=] must be nonzero.
+      1. |descriptor|.{{GPUTextureDescriptor/size}}.[=Extent3D/height=] must be nonzero.
+      1. |descriptor|.{{GPUTextureDescriptor/size}}.[=Extent3D/depth=] must be nonzero.
+      1. |descriptor|.{{GPUTextureDescriptor/mipLevelCount}} must be nonzero.
+      1. |descriptor|.{{GPUTextureDescriptor/sampleCount}} must be nonzero.
+      1. If |descriptor|.{{GPUTextureDescriptor/dimension}} is "1d":
+
+        1. |descriptor|.{{GPUTextureDescriptor/size}}.[=Extent3D/width=] must be less than or equal to 8192.
+        1. |descriptor|.{{GPUTextureDescriptor/size}}.[=Extent3D/height=] must be less than or equal to 2048.
+        1. |descriptor|.{{GPUTextureDescriptor/size}}.[=Extent3D/depth=] must be 1.
+        1. |descriptor|.{{GPUTextureDescriptor/sampleCount}} must be 1.
+
+      1. Else if |descriptor|.{{GPUTextureDescriptor/dimension}} is "2d":
+
+        1. |descriptor|.{{GPUTextureDescriptor/size}}.[=Extent3D/width=] must be less than or equal to 8192.
+        1. |descriptor|.{{GPUTextureDescriptor/size}}.[=Extent3D/height=] must be less than or equal to 8192.
+        1. |descriptor|.{{GPUTextureDescriptor/size}}.[=Extent3D/depth=] must be less than or equal to 2048.
+
+      1. Else (|descriptor|.{{GPUTextureDescriptor/dimension}} is "3d"):
+
+        1. |descriptor|.{{GPUTextureDescriptor/size}}.[=Extent3D/width=] must be less than or equal to 2048.
+        1. |descriptor|.{{GPUTextureDescriptor/size}}.[=Extent3D/height=] must be less than or equal to 2048.
+        1. |descriptor|.{{GPUTextureDescriptor/size}}.[=Extent3D/depth=] must be less than or equal to 2048.
+        1. |descriptor|.{{GPUTextureDescriptor/sampleCount}} must be 1.
+
+      1. If |descriptor|.{{GPUTextureDescriptor/sampleCount}} > 1:
+
+        1. |descriptor|.{{GPUTextureDescriptor/mipLevelCount}} must be 1.
+        1. |descriptor|.{{GPUTextureDescriptor/size}}.[=Extent3D/depth=] must be 1.
+
+      1. |descriptor|.{{GPUTextureDescriptor/mipLevelCount}} must be less than or equal to the [$maximum valid mipLevel$] for its
+         |descriptor|.{{GPUTextureDescriptor/dimension}} and |descriptor|.{{GPUTextureDescriptor/size}}.
+
+      1. If |descriptor|.{{GPUTextureDescriptor/format}} is a [=depth or stencil format=]:
+
+        1. |descriptor|.{{GPUTextureDescriptor/dimension}} must be "2d".
+        1. |descriptor|.{{GPUTextureDescriptor/sampleCount}} must be 1.
+
+      1. |descriptor|.{{GPUTextureDescriptor/usage}} must be a combination of {{GPUTextureUsage}} values.
+
+      Issue: Need to validate |descriptor|.{{GPUTextureDescriptor/sampleCount}}.
```

Comment:
We agreed that sample count could be 1 or 4 for now.

Reply:
Resolved in latest update